src/node_report.cc

```diff
@@ -0,0 +1,921 @@
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wMonth;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wDay;
+    oss << "." << std::setfill('0') << std::setw(2) << tm_struct.wHour;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wMinute;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wSecond;
+    oss << "." << pid;
+    oss << "." << std::setfill('0') << std::setw(3) << seq.load();
+#else  // UNIX, OSX
+    oss << "." << std::setfill('0') << std::setw(4) << tm_struct.tm_year+1900;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_mon + 1;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_mday;
+    oss << "." << std::setfill('0') << std::setw(2) << tm_struct.tm_hour;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_min;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_sec;
+    oss << "." << pid;
+    oss << "." << std::setfill('0') << std::setw(3) << seq.load();
+#endif
+    oss << ".json";
+  }
+
+  filename = oss.str();
+  // Open the report file stream for writing. Supports stdout/err,
+  // user-specified or (default) generated name
+  std::ofstream outfile;
+  std::ostream* outstream = &std::cout;
+  if (filename == "stdout") {
+    outstream = &std::cout;
+  } else if (filename == "stderr") {
+    outstream = &std::cerr;
+  } else {
+    // Regular file. Append filename to directory path if one was specified
+    if (env != nullptr && options->report_directory.length() > 0) {
+      std::string pathname = options->report_directory;
+      pathname += PATHSEP;
+      pathname += filename;
+      outfile.open(pathname, std::ios::out | std::ios::binary);
+    } else {
+      outfile.open(filename, std::ios::out | std::ios::binary);
+    }
+    // Check for errors on the file open
+    if (!outfile.is_open()) {
+      if (env != nullptr && options->report_directory.length() > 0) {
+        std::cerr << std::endl
+                  << "Failed to open Node.js report file: " << filename
+                  << " directory: " << options->report_directory
+                  << " (errno: " << errno << ")" << std::endl;
+      } else {
+        std::cerr << std::endl
+                  << "Failed to open Node.js report file: " << filename
+                  << " (errno: " << errno << ")" << std::endl;
+      }
+      return;
+    } else {
+      std::cerr << std::endl
+                << "Writing Node.js report to file: " << filename << std::endl;
+    }
+  }
+
+  // Pass our stream about by reference, not by copying it.
+  std::ostream& out = outfile.is_open() ? outfile : *outstream;
+
+  WriteNodeReport(
+      isolate, env, message, location, filename, out, stackstr, &tm_struct);
+
+  // Do not close stdout/stderr, only close files we opened.
+  if (outfile.is_open()) {
+    outfile.close();
+  }
+
+  std::cerr << "Node.js report completed" << std::endl;
+  if (!name.empty()) {
+    name = filename;
```

Comment:
What's this for? If the idea was to put `filename` in `name` then I think this won't work as `name` is a copy of what was passed to the function. Perhaps `name` argument was meant to be `std::string& name`?

Also, I'd prefer to have it as `std::string* name` and replace the checks appropriately as this will allow passing `nullptr` to the function without creating additional empty `std::string` every time.

Reply:
That's even better :+1:.

Reply:
Also, I think there is a mistake in the usage of return type of that function.
You have `std::string` as a return type but use `return nullptr` which shouldn't work. I guess that should be `return ""`.

Reply:
@lundibundi - thanks, fixed.